src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -35,13 +33,69 @@ public PeWritingException(Exception inner)
 
     internal static class PeWriter
     {
+        internal struct EmitBuilders
```

Comment:
Know there was, rightfully, confusion in previous iterations of the change on whose responsibility it was to free values and whether or not that was handled correctly in the code. Tried to respond to that feedback by moving all of the responsibility for the builder ownership into this type. Hoping this makes everything clearer.